modules/kafka/kafka-dest-driver.h

```diff
@@ -35,27 +35,35 @@ typedef struct
   LogTemplateOptions template_options;
   LogTemplate *key;
   LogTemplate *message;
+  LogTemplate *topic_name;
+  GHashTable *topic_hash_map;
+  GMutex lock;
 
-  gchar *topic_name;
   GList *config;
   gchar *bootstrap_servers;
+  gchar *fallback_topic_name;
   rd_kafka_topic_t *topic;
+  rd_kafka_topic_t *fallback_topic;
   rd_kafka_t *kafka;
   gint flush_timeout_on_shutdown;
   gint flush_timeout_on_reload;
   gint poll_timeout;
+  gboolean topicname_is_a_template;
 } KafkaDestDriver;
 
-void kafka_dd_set_topic(LogDriver *d, const gchar *topic);
+void kafka_dd_set_topic(LogDriver *d, LogTemplate *topic);
+void kafka_dd_set_fallback_topic(LogDriver *d, const gchar *fallback_topic);
 void kafka_dd_merge_config(LogDriver *d, GList *props);
 void kafka_dd_set_bootstrap_servers(LogDriver *d, const gchar *bootstrap_servers);
-
+gboolean _validate_topic_name(KafkaDestDriver *self, const gchar *name);
```

Comment:
this also seems to be a public function, use a proper prefix.

the underscore as a convention is used for private (static) functions.